### ISSUE TITLE
Training with known segmentation labels - error oscilations- Iterations

### ISSUE THREAD

**Issue by theoVag** [voxelmorph/voxelmorph#134]
Hello,
I have three more questions about the voxelmorph. The first one is for "Registration With Auxiliary Data" (as referenced to Voxelmorph- A learning framework for Deformable Medical Image registration). Is there an implementation for training with known segmentation labels in the github source (with Lseg loss from paper)?

The second one concerns the learning error curve that is oscillating near a small value as you can see below. I tried different lamdas (cvpr model with mse) without any chance. Is there any idea why this is happening?
![cvprloss_slurm-109846](https://user-images.githubusercontent.com/32677031/70379010-54336380-1930-11ea-9575-4bdff3ad2ec4.png)

Finally, I need a clarification for training. I understand that at an epoch the training concerns a 3d volume (for batch size=1). Could you specify what happens at every iteration inside each epoch?

Thanks a lot!

**Comment by adalca** [comment]
Hi @theoVag 

 - @balakg can you move a simple version of `train_supervised.py`?  
 - The curve looks fine to me, and doesn't seem converged (seems like it's going a bit down at the end especially). I'm not quite sure what your question is.
 - an epoch has several epochs (specified as a parameter to train.py), and each iteration feeds `batch_size` number of images to the training.

**Comment by theoVag** [comment]
I know that you are busy but a supervised train will be very helpful for my testings. Is it possible to upload the train_supervised.py? 
Thanks a lot

**Comment by adalca** [comment]
@theoVag maybe take a look at https://github.com/voxelmorph/voxelmorph/blob/redesign/scripts/tensorflow/train_semisupervised.py
